src/main/scala/com/azavea/modellab/Node.scala

```diff
@@ -15,96 +15,162 @@ import scala.collection.mutable
+    val tileRDD = FocalOperation(rasterRDD, n, Some(bounds))(op)
+      .filter { case (key, _) => bounds.contains(key.col, key.row) } // filter buffer tiles, they contain no information
+    new RasterRDD(tileRDD, metaData)
   }
 }
 
-case class LocalAdd(a: Node, b: Node) extends Node {
+case class AspectOp(
+  op: (Tile, Neighborhood, Option[GridBounds], CellSize) => Tile,
+  input: Node,
+  n: Neighborhood
+) extends Node {
   def calc(zoom: Int, bounds: GridBounds) = {
-    a(zoom, bounds).join(b(zoom, bounds)).map { case (key, (tile1, tile2)) =>
-      key -> Add(tile1, tile2)
-    }
+    val rasterRDD = input(zoom, bounds)
+    val metaData = rasterRDD.metaData
+    val cs = metaData.layout.rasterExtent.cellSize
+    val tileRDD = rasterRDD map { case (key, tile) => key -> op(tile, n, None, cs) }
+    new RasterRDD(tileRDD, metaData)
   }
 }
 
-case class LocalSubtract(a: Node, b: Node) extends Node {
+case class SlopeOp(
+  op: (Tile, Neighborhood, Option[GridBounds], CellSize, Double) => Tile,
+  input: Node,
+  n: Neighborhood,
+  z: Double
+) extends Node {
   def calc(zoom: Int, bounds: GridBounds) = {
-    a(zoom, bounds).join(b(zoom, bounds)).map { case (key, (tile1, tile2)) =>
-      key -> Subtract(tile1, tile2)
-    }
+    val rasterRDD = input(zoom, bounds)
+    val metaData = rasterRDD.metaData
+    val cs = metaData.layout.rasterExtent.cellSize
+    val tileRDD = rasterRDD map { case (key, tile) => key -> op(tile, n, None, cs, z) }
+    new RasterRDD(tileRDD, metaData)
   }
 }
 
-case class LocalMultiply(a: Node, b: Node) extends Node {
+case class LocalUnaryOp(
+  op: Function1[Tile, Tile],
+  input: Node
+) extends Node {
   def calc(zoom: Int, bounds: GridBounds) = {
-    a(zoom, bounds).join(b(zoom, bounds)).map { case (key, (tile1, tile2)) =>
-      key -> Multiply(tile1, tile2)
-    }
+    val rasterRDD = input(zoom, bounds)
+    val metaData = rasterRDD.metaData
+    val tileRDD = rasterRDD map { case (key, tile) => key -> op(tile) }
+    new RasterRDD(tileRDD, metaData)
+  }
+}
+
+case class LocalBinaryOp(
+  op: LocalTileBinaryOp,
+  input: Seq[Node],
+  const: Option[Int] = None // Constant value
+) extends Node {
+  def calc(zoom: Int, bounds: GridBounds) = {
+    val metaData = input.head(zoom, bounds).metaData
+    val tileRDD = input
+      .map { _(zoom, bounds).tileRdd }
+      .reduce { _ union _ }
+      .combineByKey(  // For local ops over tiles
+        (init: Tile) => init,
+        (aggr: Tile, value: Tile) => op(aggr, value),
+        (aggr: Tile, value: Tile) => op(aggr, value)
+      ) map { case (k, t) =>  // For constant values
+        k -> (const match {
+          case Some(constant) => { op(t, constant) }
+          case _ => t
+        })
+      }
+
+    new RasterRDD(tileRDD, metaData)
   }
 }
 
-case class LocalDivide(a: Node, b: Node) extends Node {
+case class MappingOp(
+  input: Node,
+  mapFrom: Seq[Seq[Int]],
```

Comment:
It would be very useful if this would map to NoData:

scala> "[ [[1,2,3,4], 3], [[6,8,19], null] ]".parseJson.convertTo[Array[(Array[Int], Option[Int])]]
res5: Array[(Array[Int], Option[Int])] = Array((Array(1, 2, 3, 4),Some(3)), (Array(6, 8, 19),None))